### ISSUE TITLE
App crashing when calling addStrongRef on a Java worker thread

### ISSUE THREAD

**Issue by BlueRiverInteractive** [robovm/robovm#687]
The app crashes when executing following simple code (tested in didFinishLaunching):

```Java
       new Thread(new Runnable() {
            @Override
            public void run () {
                NSObject object = new NSObject();
                NSObject ref = new NSObject();
                object.addStrongRef(ref); // <- CRASH
            }
        }).run();
```

The log shows following message:
`Cannot detach thread when there are non native frames on the call stack`

**Comment by ntherning** [comment]
I ran your code but don't get the detach error. Changing `.run()` to `.start()` however does give me the detach error.

Here's another one which crashes in the same way:

```java
import org.robovm.apple.dispatch.DispatchQueue;

public class Main {
    public static void main(String[] args) throws Exception {
        new Thread() {
            public void run() {
                DispatchQueue.getMainQueue().async(new Runnable() {
                    public void run() {
                        System.out.println("Foo bar");
                    }
                });
            }
        }.start();
    }
}
```

This is related to the fix in #652. A call to `setThreadTLS()` has to be done in `startThreadEntryPoint()`. I will take care of it.

**Comment by jtulach** [comment]
I've run into the same problem as well. Just for the record: 1.0.0.-beta-01 was OK. 1.0.0.-beta-02 was broken. My own today's build at 343b0641c2634 seems to be OK too.
